### PR TITLE
Implement wait-on support direclty in dev-pm

### PR DESCRIPTION
**Pull request by nsams**
depends on #23 

Main benefit over using the wait-on direclty in script: we can show 'Waiting' in status output.

(this PR also does a big refactor and introduces a Script class)

### CODE REVIEW

**Review by Fabian-Fynn — APPROVED** [review]

File: src/daemon-command/script.ts [hunk 1]
```diff
@@ -37,18 +49,54 @@ export class Script {
...
-        console.log(`${colors.bgGreen.bold.black(" dev-pm ")} starting: ${this.scriptDefinition.script}`);
+    async startProcess(): Promise<void> {
+        if (this.waitOn.length > 0) {
+            this.status = "waiting";
+
+            try {
+                //first a silent try for 200ms (to avoid too much console output)
+                await waitOn({
+                    resources: this.waitOn,
+                    timeout: 1000,
```
Comment (Fabian-Fynn) [inline src/daemon-command/script.ts:73]: shouldn't this be 200?

Comment (Fabian-Fynn) [reply src/daemon-command/script.ts:73]: Or did you mean to add a `delay: 200`?

Comment (nsams) [reply src/daemon-command/script.ts:73]: yes, the comment is outdated, removed the value

(no, delay does something else)

Comment (Fabian-Fynn) [reply src/daemon-command/script.ts:73]: ok but then the logs below are also wrong

Comment (nsams) [reply src/daemon-command/script.ts:73]: oops

**Review by Fabian-Fynn — APPROVED** [review]

**Pull request merged**